packages/p2p/src/components/my-profile/payment-methods/payment-methods-list/edit-payment-method-form.jsx

```diff
@@ -144,8 +146,12 @@ const EditPaymentMethodForm = () => {
                                     secondary
                                     large
                                     onClick={() => {
-                                        my_profile_store.setShouldShowEditPaymentMethodForm(false);
-                                        my_profile_store.setPaymentMethodToEdit();
+                                        if (dirty) {
+                                            my_profile_store.setIsCancelEditPaymentMethodModalOpen(true);
+                                        } else {
+                                            my_profile_store.setPaymentMethodToEdit(null);
+                                            my_profile_store.setShouldShowEditPaymentMethodForm(false);
+                                        }
```

Comment:
When user has not made any changes to the edit form, Cancel button should not display the Cancel Edit Confirmation modal as per QA requirements